### PR TITLE
allow xtask install --client[=CLIENT] to specify client

### PR DESCRIPTION
**Pull request by Emilgardis**

### CODE REVIEW

File: xtask/src/main.rs [hunk 1]
```diff
@@ -46,10 +46,11 @@ USAGE:
     cargo xtask install [FLAGS]
 
 FLAGS:
-        --client-code    Install only VS Code plugin
-        --server         Install only the language server
-        --mimalloc       Use mimalloc for server
-    -h, --help           Prints help information
+        --client-code[=CLIENT]    Install only VS Code plugin.
+                                  CLIENT is 'code', 'code-insiders', 'codium', or 'code-oss'
```
Comment (bjorn3) [inline xtask/src/main.rs:50]:
```suggestion
                                  CLIENT is one of 'code', 'code-insiders', 'codium', or 'code-oss'
```

File: xtask/src/install.rs [hunk 2]
```diff
@@ -74,17 +109,13 @@ fn fix_path_for_mac() -> Result<()> {
...
     let find_code = |f: fn(&str) -> bool| -> Result<&'static str> {
-        ["code", "code-insiders", "codium", "code-oss"]
-            .iter()
-            .copied()
-            .find(|bin| f(bin))
-            .ok_or_else(|| {
-                format_err!("Can't execute `code --version`. Perhaps it is not in $PATH?")
-            })
+        client_opt.as_cmds().iter().copied().find(|bin| f(bin)).ok_or_else(|| {
+            format_err!("Can't execute `code --version`. Perhaps it is not in $PATH?")
```
Comment (bjorn3) [inline xtask/src/install.rs:117]: When using `ClientOpt::All` it should attempt to install for all found vscode variants I think. Or it should be renamed to `ClientOpt::Any`.

Comment (Emilgardis) [reply xtask/src/install.rs:117]: That is not what xtask install --client-code does today. I think that's outside the scope of this PR.

I'll change the name though to Any

File: xtask/src/main.rs [hunk 3]
```diff
@@ -46,10 +46,11 @@ USAGE:
     cargo xtask install [FLAGS]
 
 FLAGS:
-        --client-code    Install only VS Code plugin
-        --server         Install only the language server
-        --mimalloc       Use mimalloc for server
-    -h, --help           Prints help information
+        --client-code[=CLIENT]    Install only VS Code plugin.
```
Comment (matklad) [inline xtask/src/main.rs:49]:
```suggestion
        --client [=CLIENT]    Install only VS Code plugin.
```

we never had any other clients, so it's better to simplify this while we are at it.

**Comment by matklad** [comment]
r=me with option renamed!

bors d+

File: xtask/src/main.rs [hunk 4]
```diff
@@ -46,19 +46,20 @@ USAGE:
     cargo xtask install [FLAGS]
 
 FLAGS:
-        --client-code    Install only VS Code plugin
-        --server         Install only the language server
-        --mimalloc       Use mimalloc for server
-    -h, --help           Prints help information
+        --client[=CLIENT] Install only VS Code plugin.
```
Comment (Emilgardis) [inline xtask/src/main.rs:49]: uncertain if this should have more spaces, I'll leave as this, seems fine

**Pull request merged**